File: go/guardnet.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// GuardNetService — V3 advanced defense (GuardNet, JBShield, CCFC, MULI).
type GuardNetService struct {
	t *internal.Transport
}

func (s *GuardNetService) post(ctx context.Context, path, text string) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", path, map[string]any{"text": text}, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *GuardNetService) Analyze(ctx context.Context, text string) (map[string]any, error) {
	return s.post(ctx, "/v3/defense/guardnet", text)
}
func (s *GuardNetService) JBShield(ctx context.Context, text string) (map[string]any, error) {
	return s.post(ctx, "/v3/defense/jbshield", text)
}
func (s *GuardNetService) CCFC(ctx context.Context, text string) (map[string]any, error) {
	return s.post(ctx, "/v3/defense/ccfc", text)
}
func (s *GuardNetService) MULI(ctx context.Context, text string) (map[string]any, error) {
	return s.post(ctx, "/v3/defense/muli", text)
}
func (s *GuardNetService) Unified(ctx context.Context, text string) (map[string]any, error) {
	return s.post(ctx, "/v3/defense/unified", text)
}

File: python/src/aegis/resources/guardnet.py
"""GuardNet and V3 defense resources — V3 advanced defense endpoints."""

from __future__ import annotations

from typing import Any, Dict

from aegis.resources._base import AsyncResource, SyncResource


class SyncGuardNet(SyncResource):

    def analyze(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/defense/guardnet", json={"text": content, **kwargs})

    def jbshield(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/defense/jbshield", json={"text": content, **kwargs})

    def ccfc(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/defense/ccfc", json={"text": content, **kwargs})

    def muli(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/defense/muli", json={"text": content, **kwargs})

    def unified(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/defense/unified", json={"text": content, **kwargs})


class AsyncGuardNet(AsyncResource):

    async def analyze(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/defense/guardnet", json={"text": content, **kwargs})

    async def jbshield(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/defense/jbshield", json={"text": content, **kwargs})

    async def ccfc(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/defense/ccfc", json={"text": content, **kwargs})

    async def muli(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/defense/muli", json={"text": content, **kwargs})

    async def unified(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/defense/unified", json={"text": content, **kwargs})
